app/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source.rb

```diff
@@ -46,14 +47,31 @@ def raw_delete_in_provider
   end
 
   def git_repository
-    super || begin
-      transaction do
-        update!(:git_repository => GitRepository.create!(:url => scm_url))
+    (super || (ensure_git_repository && super)).tap { |r| sync_git_repository(r) }
+  end
+
+  private def ensure_git_repository
+    transaction do
+      repo = GitRepository.create!(attrs_for_sync_git_repository)
+      if new_record?
+        self.git_repository_id = repo.id
+      else
+        update_columns(:git_repository_id => repo.id) # rubocop:disable Rails/SkipsModelValidations
       end
-      super
     end
   end
 
+  private def sync_git_repository(git_repository = nil)
+    return unless name_changed? || scm_url_changed? || authentication_id_changed?
```

Comment:
FYI I really tried to DRY this up with respect to the `attrs_for_sync_git_repository`, by using something like `changed.include_any?`, but because one side has `:url` and the other side has `:scm_url` it got really ugly, so I'd prefer to just leave it this way for now.